web6/web6.go

```diff
@@ -801,3 +801,21 @@ func dynamicPage_404(uri string, w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusNotFound)
 	w.Write([]byte(base_html))
 }
+
+func dynamicPage_500(error_msg string, w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "text/html")
+
+	//TODO(z): Create a formatted 500 Internal Server Error html page like dynamicPage_404 and return that instead of text
+	error_response := fmt.Sprintf("500 Internal Error. %v", error_msg)
+
+	w.WriteHeader(http.StatusInternalServerError)
+	w.Write([]byte(error_response))
+}
+
+func recoverError_500(w http.ResponseWriter, r *http.Request) {
+	// Recover from internal panics until we could guarantee no panics
+	if recover := recover(); recover != nil {
```

Comment:
Might need to check here that the app hasn't started writing down the ResponseWriter

Reply:
Is there some way to clear ResponseWriter? Did some research but couldn't find a method to clear what has already been written to ResponseWriter. Currently, the last thing that the handler does is write to the ResponseWriter (so panics are before writing to ResponseWriter)

However, in the future, if there is a panic after we write to ResponseWriter, we would get a weird page where the 500 error is at the bottom with previous written content on top